### PR TITLE
Add dated-folder extension

### PR DESCRIPTION
**Pull request by fanhefeng**
## Description

**Dated Folder** adds one `no-view` command, **Create Dated Folder**: it creates a folder named after today's date (e.g. `2026-08-28`) inside a parent folder of your choice and opens it straight in a terminal. It is meant for the "I need somewhere to put today's scratch work" moment — one keystroke instead of `mkdir` + `cd` + opening a terminal window.

Preferences:

- **Parent Folder** (`directory`) — defaults to `~/Desktop/temp` when empty.
- **Terminal** (`appPicker`) — any installed app. When empty, the extension follows the terminal registered in LaunchServices as the shell-role handler for `public.unix-executable`, which is exactly what Ghostty / iTerm2 / Warp write when you accept "make this your default terminal". Falls back to Terminal.app.
- **Folder Name Format** (`textfield`) — `yyyy`, `MM`, `dd`, `HH`, `mm`, `ss` tokens; `/` nests folders (`yyyy/MM/dd`). Names containing `..` segments are rejected so a format string can never escape the parent folder.
- **Reveal** (`checkbox`) — also reveal the folder in Finder.

No network access, no binaries, no third-party dependencies beyond `@raycast/api` / `@raycast/utils`.

`platforms` is `["macOS"]` only. A Windows launcher is already in `src/lib.ts` (Windows Terminal via `wt.exe -d`, other shells via `start /d`, Desktop resolved from the known-folder registry entry so OneDrive redirection is respected), but it has not been exercised on a real Windows machine yet, so the extension does not claim support it cannot back. Windows will be enabled in a follow-up PR after manual QA.

## Screencast

The command has no UI of its own (`no-view`): it creates the folder, opens the terminal, and confirms with a HUD such as `📂 Ghostty → ~/Desktop/temp/2026-08-28`.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used in the `README` are located outside the metadata folder if they were not generated with our [metadata tool](https://developers.raycast.com/basics/prepare-an-extension-for-store#how-to-use-it)

### CODE REVIEW

File: extensions/dated-folder/src/create-dated-folder.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  isWindows,
+  openTerminalWindows,
+} from "./lib";
+
+interface Preferences {
+  parentDir?: string;
+  terminal?: Application;
+  dateFormat?: string;
+  revealInFinder: boolean;
+}
```
[the remark this answers is not in the document]
Comment (fanhefeng) [reply extensions/dated-folder/src/create-dated-folder.ts]: Valid — fixed in `aee14773`. The hand-written interface is gone; the command now reads `getPreferenceValues<Preferences.CreateDatedFolder>()` from the generated `raycast-env.d.ts`, so the manifest is the only place the preference contract lives.

File: extensions/dated-folder/src/lib.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+ */
+export async function defaultTerminalBundleId(): Promise<string> {
+  if (!existsSync(LS_PREFS)) return FALLBACK_TERMINAL_BUNDLE;
+  try {
+    const { stdout } = await execFileAsync("/usr/bin/plutil", ["-convert", "json", "-o", "-", LS_PREFS]);
+    const handlers: unknown = JSON.parse(stdout).LSHandlers;
+    if (!Array.isArray(handlers)) return FALLBACK_TERMINAL_BUNDLE;
+    const entry = handlers.find(
+      (h) => h?.LSHandlerContentType === "public.unix-executable" && typeof h?.LSHandlerRoleShell === "string",
+    );
```
[the remark this answers is not in the document]
Comment (fanhefeng) [reply extensions/dated-folder/src/lib.ts]: Valid — fixed in `aee14773`. `currentShellHandler()` now collects every `public.unix-executable` record that carries a shell-role handler and follows the one with the newest `LSHandlerModificationDate`, i.e. the terminal the user chose most recently, instead of whichever record happens to come first in the file. Unit-checked with mixed-order and undated records. The uninstalled-bundle case was already covered: the bundle id is matched against `getApplications()` and the command falls back to Terminal.app when it is not installed.

File: extensions/dated-folder/src/lib.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+/**
+ * macOS has no "default terminal" setting in System Settings, but LaunchServices stores the
+ * shell-role handler for `public.unix-executable`. That is exactly what terminal apps write when
+ * they ask "make this your default terminal?" on first launch, so it is the value to follow.
+ */
+export async function defaultTerminalBundleId(): Promise<string> {
+  if (!existsSync(LS_PREFS)) return FALLBACK_TERMINAL_BUNDLE;
+  try {
+    const { stdout } = await execFileAsync("/usr/bin/plutil", ["-convert", "json", "-o", "-", LS_PREFS]);
+    return currentShellHandler(JSON.parse(stdout).LSHandlers) ?? FALLBACK_TERMINAL_BUNDLE;
```
[the remark this answers is not in the document]
Comment (fanhefeng) [reply extensions/dated-folder/src/lib.ts:84]: Not reproducible — `LSHandlerModificationDate` is stored as a `real` (CFAbsoluteTime, seconds since 2001), not a plist `date`, so `plutil -convert json` handles the file. Checked on a machine with 62 `LSHandlerModificationDate` entries: the conversion exits 0, every value comes back as a JSON number, and the shell-role handler resolves to the terminal the user actually chose (Ghostty). The `catch` only covers a corrupt or unreadable plist, and even then the command still opens the folder — in Terminal.app — rather than failing after the folder was created.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @fanhefeng 👋

Thanks for your contribution 💪

In `src/lib.ts`, `resolveInside` rejects every target when Parent Folder is `/`, because `base + sep` becomes `//`. Could you handle an existing trailing separator and add a regression test?

```ts
const prefix = base.endsWith(sep) ? base : base + sep;
return target.startsWith(prefix) ? target : null;
```

For example, `resolveInside("/", "2026-09-07")` should return `/2026-09-07`.

Small cleanup: the Windows launcher is unreachable with the current macOS-only manifest. I appreciate you not claiming untested support. Could you move it to the Windows follow-up PR or add a clear comment above that section saying it’s staged for that release?

---

I’ll convert this PR into a draft after submitting this review. Please press Ready for review when it’s ready and we’ll have a look 😊

**Comment by fanhefeng** [comment]
Thanks for the careful read, @pernielsentikaer — both handled in `a01fe389`.

**1. `resolveInside` with `/` as the parent.** Confirmed and fixed. One extra note on your snippet: dropping the trailing separator also drops the guard that kept the parent *itself* out. With a root parent the prefix **is** the parent, so `resolveInside("/", "..")` returned `/` instead of `null` — the command would then have opened a terminal at `/` rather than reporting the bad format. The regression test caught it on the first run, so the check now reads:

```ts
const prefix = base.endsWith(sep) ? base : base + sep;
return target !== base && target.startsWith(prefix) ? target : null;
```

`resolveInside("/", "2026-09-07")` → `/2026-09-07`, and `resolveInside("/", "..")` → `null`.

**2. Regression tests.** Added `tests/lib.test.js` — 21 cases over `formatDate`, `resolveInside`, `tildify`, and `currentShellHandler`. It follows the setup already merged in `file-tidy`: Node's built-in test runner via `npm test`, **no new dependencies**, so nothing changes for `ray build`, `ray lint`, or the Socket/NPM checks. The root-parent case and the `..` case above are both in there, plus the boundary cases around them (`..foo` is a legitimate folder name and must be *allowed*; `../tmpevil` next to `/tmp` must be *rejected*).

**3. Windows launcher.** Removed — `openTerminalWindows`, `launch`, the Windows Terminal alias lookup, and the `reg.exe` Desktop resolution are all gone, along with the `isWindows` branch and the Windows notes in the README. `defaultParentDir` is synchronous again as a result. That is ~110 lines out, and the extension is now macOS-only in code as well as in the manifest. The implementation is preserved in the branch history and will come back in the Windows PR, verified on a real machine.

**One unrelated fix I found while writing the tests:** the HUD shortened the path with `target.replace(homedir(), "~")`, a bare substring match — a parent folder like `/Users/anna-old/…` displayed as `~-old/…`. It now compares whole path segments (`tildify`), with tests.

`ray build`, `ray lint`, `tsc --noEmit`, and all 21 tests pass. Marking as ready for review 🙏

**Review by pernielsentikaer — APPROVED** [review]
Looks good to me, approved 🔥
